### ISSUE TITLE
Add-on terminates/quits/crashes transmission-daemon

### ISSUE THREAD

**Issue by Kihltech** [Mika-/torrent-control#8]
(In reply to https://addons.mozilla.org/en-US/firefox/addon/torrent-control/reviews/996230/)

**My setup:**
- Ubuntu Server 16.04 running headless **transmission-daemon**: 2.92-1ubuntu1~16.104.2. 
RPC port open to clients on local network.
- Firefox addon on multiple clients 

Installing this addon and adding a torrent from the context menu terminates/quits/crashes transmission-daemon.

**syslog:**
```Dec  3 13:13:00 ubuntu-server transmission-daemon[11338]: Closing transmission session... done.```

**transmission-daemon debug-log:**
```[2017-12-03 13:44:21.497] ubuntu-17.10-desktop-amd64.iso Queued for verification (verify.c:269)
[2017-12-03 13:44:21.497] ubuntu-17.10-desktop-amd64.iso Verifying torrent (verify.c:224)
[2017-12-03 13:44:21.497] ubuntu-17.10-desktop-amd64.iso verifying torrent... (verify.c:62)
[2017-12-03 13:44:21.497] ubuntu-17.10-desktop-amd64.iso Verification is done. It took 0 seconds to verify 1501102080 bytes (1501102080 bytes per second) (verify.c:167)
[2017-12-03 13:44:21.497] ubuntu-17.10-desktop-amd64.iso Announcing to tracker (announcer.c:1552)
[2017-12-03 13:44:21.497] ubuntu-17.10-desktop-amd64.iso Announcing to tracker (announcer.c:1552)
[2017-12-03 13:44:24.301] Saved "/home/transmission/.config/transmission-daemon/settings.json" (variant.c:1266)
[2017-12-03 13:44:24.301] DHT Uninitializing DHT (tr-dht.c:348)
[2017-12-03 13:44:24.301] DHT Not saving nodes, DHT not ready (tr-dht.c:359)
[2017-12-03 13:44:24.301] DHT Done uninitializing DHT (tr-dht.c:396)
[2017-12-03 13:44:24.301] Port Forwarding Stopped (port-forwarding.c:180)
[2017-12-03 13:44:24.301] RPC Server Stopped listening on 0.0.0.0:9091 (rpc-server.c:801)
[2017-12-03 13:44:24.301] ubuntu-17.10-desktop-amd64.iso Removing torrent (torrent.c:1915)
[2017-12-03 13:44:24.301] ubuntu-17.10-desktop-amd64.iso Pausing (torrent.c:1856)
[2017-12-03 13:44:24.301] Saved "/home/transmission/.config/transmission-daemon/resume/ubuntu-17.10-desktop-amd64.iso.40448d478d9203a3.resume" (variant.c:1266)
[2017-12-03 13:44:24.301] Saved "/home/transmission/.config/transmission-daemon/stats.json" (variant.c:1266)
[2017-12-03 13:44:24.301] Closing libevent thread (trevent.c:262)```

As you can see, a couple of seconds after the torrent file is added, the daemon starts shutting down...

Thanks for you quick feedback before. I'm happy to help debugging and troubleshooting in order to get it working.

- Robert

**Comment by Mika-** [comment]
That really is a problem. Looks like "transmission session" in a documentation refers to actual application session and not to a remote session token. I have just released a fix for this in 0.1.8 (you can manually force an update in Add-ons > Extensions > *Gear*> Check for updates). Please comment below how that is working for you.

My test clients are running in a docker containers as a daemon. And if run that way Transmission apparently just silently discards session-close action.

**Comment by Kihltech** [comment]
Excellent! Now it works as I assume is intended, no more shutdown of the daemon.

That was really quickly fixed!
